packages/editor-ui/src/components/Node.vue

```diff
@@ -340,10 +356,27 @@ export default mixins(
 			isTouchActive: false,
 			nodeSubtitle: '',
 			showTriggerNodeTooltip: false,
+			showPinDataDiscoveryTooltip: false,
 			dragging: false,
 		};
 	},
 	methods: {
+		checkPinDataDiscoveryFlow(dataItemsCount: number): void {
+			if (!this.isTriggerNode) {
+				return;
+			}
+
+			const hasSeenPinDataTooltip = localStorage.getItem(LOCAL_STORAGE_PIN_DATA_INITIAL_DISCOVERY_WORKFLOW_FLAG);
+
+			if (dataItemsCount > 0 && !hasSeenPinDataTooltip) {
+				localStorage.setItem(LOCAL_STORAGE_PIN_DATA_INITIAL_DISCOVERY_WORKFLOW_FLAG, 'true');
+
+				this.showPinDataDiscoveryTooltip = true;
+				setTimeout(() => {
+					this.showPinDataDiscoveryTooltip = false;
+				}, 5000);
+			}
+		},
```

Comment:
1. For learning, why local storage instead of a Vuex store value coming from the `settings` column in the `user` table? Do we want to make it easy to keep this hint showing every time local storage is cleared?
2. Ideally, `dataItemsCount` integer should be checked before querying local storage.
3. `LOCAL_STORAGE_PIN_DATA_INITIAL_DISCOVERY_WORKFLOW_FLAG` Should we specify the medium `LOCAL_STORAGE` in the constant name? A generic name would prevent us from having to rename when refactoring later.
4. `LOCAL_STORAGE_PIN_DATA_INITIAL_DISCOVERY_WORKFLOW_FLAG` Does `INITIAL` add anything here? Discovery happens only once, at least until local storage cleared.
5. `LOCAL_STORAGE_PIN_DATA_INITIAL_DISCOVERY_WORKFLOW_FLAG` I find `CANVAS` instead of `WORKFLOW` more accurate + descriptive.
6. `hasSeenPinDataTooltip`, `showPinDataDiscoveryTooltip`, etc. Should we mention the medium `Tooltip`? A generic name like `Hint` prevents us from having to rename based on later refactorings, as had to be done for `.edit-mode-footer-buttons` → `.edit-mode-actions`
7. `checkPinDataDiscoveryFlow` The main purpose of this method is updating local state to show the hint, the checks are only there to make sure this happens in specific conditions. `showPinDataHint`?
8. Matter of preference, L365-7 could be a single line.
9. `5000` In my experience, 5 seconds was not nearly enough, because you first have to realize that there is a new tooltip to read, which is easy to miss in the NDV and especially when you're alt-tabbing back into the NDV from where you made the request to manually trigger the trigger node. Also the wording is not simple enough for a flashing tooltip: `You can pin this data to avoid having to generate an event every time.` Maybe 10 seconds with `You can pin the data received by this node [Learn more]` with tooltip kept open on hover.

Reply:
1. Agreed upon this with Mutasem and David. It was the easiest choice from an effort/result point of view.
2. Updated flow.
3. I think it's useful to know where this key is being used. We have `LOCAL_STORAGE_ACTIVATION_FLAG` following the same naming pattern.
4. Removed `INITIAL`.
5. Replaced `WORKFLOW` with `CANVAS`.
6. From a component / ui terminology perspective, `Tooltip` and `Hint` are two different things. We're essentially using these to control the visibility of a tooltip, which is why I named the variables like this. 
7. For clarity, renamed visibility variable to `pinDataDiscoveryTooltipVisible` and method to `showPinDataDiscoveryTooltip`.
9. Will ask David about the message. Updated time to 10s meanwhile.